### PR TITLE
drivers: added driver for ina226 current monitor

### PR DESCRIPTION
**Pull request by MichelRottleuthner**
This is a simple device driver for the ina226 current and power monitor using I2C. The code structure is based on the existing driver for ina220. The included test application was tested on samR21-xpro. Instructions on how to use this driver can be found in tests/driver_ina226/README.md.

### CODE REVIEW

**Comment by MichelRottleuthner** [comment]
>Would it be feasible to combine the ina220 and ina226 drivers into one ina2xx, or are they very different?

That's probably feasible. The general procedure is very similar with both drivers (just reading and writing some registers). There are things that would need to be separated though (there is a programmable gain amplifier in the ina220 that isn't there in the ina226 while the ina226 has some alert functionality that isn't supported by the ina220). Also the configuration parameters like averaging and conversion times are not compatible between both devices. I need to look into the details of the ina220 data sheet to see if maybe there are more pitfalls.

**Comment by PeterKietzmann** [comment]
Unfortunately we still don't have a general procedure to merge similar sensor drivers (compare #5484 and #5433). Any suggestions @gebart? As long as differences are small it shouldn't be too hard though. @MichelRottleuthner I'm  looking forward for your detail-report ;-) before reviewing

**Comment by MichelRottleuthner** [comment]
Both drivers -at least for now- don't do any (model specific) interpretation of the register values ie. there is no translation from raw register values to actual V, mA etc. scale (something like that would be nice from the usability point of view, though). For now this job is up to the user. So the basic things that are supported by both chips can be done by the same driver function. 
But what to do with the model specific stuff? 
I could imagine simply putting the model specific structs like [ina220_range](https://github.com/RIOT-OS/RIOT/blob/master/drivers/include/ina220.h#L56), [ina226_vbusct](https://github.com/MichelRottleuthner/RIOT/blob/ina226devel/drivers/include/ina226.h#L93) etc. into the same file (eg. ina22x.h) while keeping their original names to make it clear to the user that these settings are only meant to be used with either the ina220 or the ina226.
This approach of keeping the names of things that are compatible with only one model and merging equivalent ones into ina22x* could also be done with the functions albeit with the implication that the user takes the responsibility to call only the appropriate functions. 
Another thing that could be done is to use some `#ifdef` switching to choose between 220/226 support. This comes with the -maybe negligible- downside that you can use only one specific model in your system at a time.
That leaves the use of a sensor type parameter like it is used in the generic [dht](https://github.com/RIOT-OS/RIOT/blob/master/drivers/include/dht.h#L51) driver.
What are you guys @gebart @PeterKietzmann thinking?

File: drivers/ina226/ina226.c [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+int ina226_init(ina226_t *dev, i2c_t i2c, uint8_t address)
+{
+    /* write device descriptor */
+    dev->i2c = i2c;
+    dev->addr = address;
+    return 0;
+}
+
+int ina226_read_calibration(ina226_t *dev, uint16_t *calibration)
```
Comment (cgundogan) [inline drivers/ina226/ina226.c]: all those one-line functions could be moved to the header file and marked as `inline`

File: drivers/ina226/include/ina226-regs.h [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     driver_ina226
+ * @{
+ *
+ * @file
+ * @brief       Register definitions for Texas Instruments INA226 High-Side or Low-Side Measurement,
+ *                 Bi-Directional Current and Power Monitor with I2C Compatible Interface
```
Comment (cgundogan) [inline drivers/ina226/include/ina226-regs.h]: indent

**Comment by OlegHahm** [comment]
I think this PR could need some further discussion.

**Comment by PeterKietzmann** [comment]
@MichelRottleuthner I appreciate your great work on that device driver. However, IIRC you mentioned a re-write some time ago. Does that mean we can close this PR?

**Comment by PeterKietzmann** [comment]
As discussed offline I will close this PR now with memo label set. An alternative will follow at some point in time.

**Pull request closed without merging**